src/app/rpmostree-compose-builtin-tree.c

```diff
@@ -1034,13 +1047,109 @@ impl_compose_tree (const char      *treefile_pathstr,
 
   g_print ("%s => %s\n", self->ref, new_revision);
 
+  /* Finally process the --touch-if-changed option  */
   if (!process_touch_if_changed (error))
     return FALSE;
 
   return TRUE;
 }
 
 int
+rpmostree_compose_builtin_install_root (int             argc,
+                                        char          **argv,
+                                        RpmOstreeCommandInvocation *invocation,
+                                        GCancellable   *cancellable,
+                                        GError        **error)
+{
+  g_autoptr(GOptionContext) context = g_option_context_new ("TREEFILE DESTDIR");
+  if (!rpmostree_option_context_parse (context,
+                                       option_entries,
+                                       &argc, &argv,
+                                       invocation,
+                                       cancellable,
+                                       NULL, NULL, NULL, NULL,
+                                       error))
+    return EXIT_FAILURE;
+
+  if (argc != 3)
+   {
+      rpmostree_usage_error (context, "TREEFILE and DESTDIR required", error);
+      return EXIT_FAILURE;
+    }
+
+  if (!opt_repo)
+    {
+      rpmostree_usage_error (context, "--repo must be specified", error);
+      return EXIT_FAILURE;
+    }
+
+  if (opt_workdir)
+    {
+      rpmostree_usage_error (context, "--workdir is ignored with install-root", error);
+      return EXIT_FAILURE;
+    }
+
+  const char *treefile_path = argv[1];
+  /* Destination is turned into workdir */
+  const char *destdir = argv[2];
+  opt_workdir = g_strdup (destdir);
+
+  g_autoptr(RpmOstreeTreeComposeContext) self = NULL;
+  if (!rpm_ostree_compose_context_new (treefile_path, &self, cancellable, error))
+    return FALSE;
+  if (!impl_install_tree (self, cancellable, error))
+   return EXIT_FAILURE;
+  /* Keep the dir around */
+  g_print ("rootfs: %s/rootfs\n", self->workdir_tmp.path);
+  glnx_tmpdir_unset (&self->workdir_tmp);
```

Comment:
Just to confirm, in the `compose tree` case, this will still be nuked by the `RpmOstreeTeeComposeContext` cleanup in `commit_root`, right?

Reply:
Yep!